Packs/rasterize/Integrations/rasterize/rasterize.py

```diff
@@ -365,39 +372,148 @@ def start_chrome_headless(chrome_port, chrome_binary=CHROME_EXE, user_options=""
+
+    info = read_info_file(CHROME_INSTANCES_FILE_PATH)
+    demisto.log(f"{info=}")  # TODO: Remove log
+
+    instance_id_to_chromes_options, chrome_options_to_port, instance_id_to_port = get_chrome_instances_info_dictionaries(info)
+
+    if info:
+        demisto.log("there is an info")  # TODO: Remove log
+        info_without_new_lines = info.strip()
+        splitted_info = info_without_new_lines.splitlines()
+        _, instances_id_tuple, chromes_options_tuple = zip(*[line.strip().split('\t') for line in splitted_info])
+        instances_id = list(instances_id_tuple) if instances_id_tuple else []
+        chromes_options = list(chromes_options_tuple) if chromes_options_tuple else []
+
+        if not chromes_options and instances_id:
+            chromes_options.append('None')
+
+        demisto.log(f"{chromes_options=}, {instances_id=}")  # TODO: Remove log
+
+    if not info or (chrome_options in chromes_options and instance_id not in instances_id) or (
+        chrome_options not in chromes_options and instance_id not in instances_id):
+        """
+        When:
+            case 1: file is empty, that means no chrome open on the machine
+            or
+            case 2: new instance with the same existing chrome configuration as another instance
+            or
+            case 3: chrome configuration does not exist and integration instance does not exist in the file
+        Then:
+            create new browser
+        """
+
+        # TODO: remove the block below
+        demisto.log("in chrome manager first if - case 1,2,3")
+        if not info:
+            demisto.log("first if - no info file found (case 1)")
         else:
-            write_info_file(PORT_FILE_PATH, '')
+            demisto.log("first if - info file exist")
+            if chrome_options in chromes_options and instance_id not in instances_id:
+                demisto.log(f"chrome_options in chromes_options and instance_id not in instances_id (case 2)")
+            elif chrome_options not in chromes_options and instance_id not in instances_id:
+                demisto.log(f"chrome_options not in chromes_options and instance_id not in instances_id (case 3)")
+
+        return generate_new_chrome_instance(instance_id, chrome_options)
+
+    elif chrome_options not in instance_id_to_chromes_options[instance_id] and instance_id in instances_id:
+        """
+        When:
+            case 4: instance updated the chrome configuration to different configuration
+        Then:
+            should kill the old browser and create new one
+        """
+
+        demisto.log(f"case 4, chrome_options not in chromes_options and instance_id in instances_id")  # TODO: Remove log
+        demisto.debug(f"Found instance_id but with different {chrome_options=}")
+        chrome_port = instance_id_to_port.get(instance_id)
+        terminate_chrome(chrome_port)
+        delete_row_with_old_chrome_configurations_from_info_file(info, instance_id, chrome_port)
```

Comment:
Why not update the file first, and then terminate the chrome?
That way, the worse that could happen if that you'll have several chrome(s) running (which can be cleaned by checking the chrome_instances.tsv file.

The way it's written now - The chrome might be terminated, but the file would have info about a non-existing chrome.

Reply:
good idea..